mso/resource_mso_schema_site_anp_epg_bulk_staticport.go

```diff
@@ -0,0 +1,749 @@
+	stateSiteId := d.Get("site_id").(string)
+	stateTemplateName := d.Get("template_name").(string)
+	stateANPName := d.Get("anp_name").(string)
+	stateEpgName := d.Get("epg_name").(string)
+
+	staticPortsList := make([]interface{}, 0, 1)
+	if staticPortsValue, ok := d.GetOk("static_ports"); ok {
+		staticPorts := staticPortsValue.([]interface{})
+		for _, staticPortValue := range staticPorts {
+			staticPort := staticPortValue.(map[string]interface{})
+			staticPortMap := make(map[string]interface{})
+			var static_port_pod, static_port_leaf, static_port_path, static_port_fex string
+
+			if staticPort["path_type"] != nil {
+				staticPortMap["type"] = staticPort["path_type"]
+			}
+			if staticPort["deployment_immediacy"] != nil {
+				staticPortMap["deploymentImmediacy"] = staticPort["deployment_immediacy"]
+			}
+			if staticPort["mode"] != nil {
+				staticPortMap["mode"] = staticPort["mode"]
+			}
+			if staticPort["vlan"] != nil {
+				staticPortMap["portEncapVlan"] = staticPort["vlan"]
+			}
+			if staticPort["micro_seg_vlan"] != 0 {
+				staticPortMap["microSegVlan"] = staticPort["micro_seg_vlan"]
+			}
+			if staticPort["pod"] != nil {
+				static_port_pod = staticPort["pod"].(string)
+			}
+			if staticPort["leaf"] != nil {
+				static_port_leaf = staticPort["leaf"].(string)
+			}
+			if staticPort["path"] != nil {
+				static_port_path = staticPort["path"].(string)
+			}
+			if staticPort["fex"] != nil {
+				static_port_fex = staticPort["fex"].(string)
+			}
+
+			var portpath string
+
+			if staticPortMap["path_type"] == "port" && staticPort["fex"] != "" {
+				portpath = fmt.Sprintf("topology/%s/paths-%s/extpaths-%s/pathep-[%s]", static_port_pod, static_port_leaf, static_port_fex, static_port_path)
+			} else if staticPortMap["path_type"] == "vpc" {
+				portpath = fmt.Sprintf("topology/%s/protpaths-%s/pathep-[%s]", static_port_pod, static_port_leaf, static_port_path)
+			} else {
+				portpath = fmt.Sprintf("topology/%s/paths-%s/pathep-[%s]", static_port_pod, static_port_leaf, static_port_path)
+			}
+
+			staticPortMap["path"] = portpath
+
+			staticPortsList = append(staticPortsList, staticPortMap)
+		}
+	}
+
+	site, err := getSiteFromSiteIdAndTemplate(schemaId, stateSiteId, stateTemplateName, msoClient)
+	if err != nil {
+		return err
+	} else {
+		d.Set("site_id", stateSiteId)
+		d.Set("template_name", stateTemplateName)
+	}
+
+	anpCont, err := getSiteAnp(stateANPName, site)
+	if err != nil {
+		return err
+	} else {
+		d.Set("anp_name", stateANPName)
+	}
+
+	_, err = getSiteEpg(stateEpgName, anpCont)
+	if err != nil {
+		return err
+	} else {
+		d.Set("epg_name", stateEpgName)
+	}
+
+	pathsp := fmt.Sprintf("/sites/%s-%s/anps/%s/epgs/%s/staticPorts", stateSiteId, stateTemplateName, stateANPName, stateEpgName)
+	staticStruct := models.NewSchemaSiteAnpEpgBulkStaticPort("replace", pathsp, staticPortsList)
+	_, errs := msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", schemaId), staticStruct)
```

Comment:
Just to validate I understand correct, we PATCH replace the entire list and thus do not need any delete logic for items that are not in the tf configuration but are part of ndo configuration?

Reply:
yes